include/common/intops.rs

```diff
@@ -41,9 +41,19 @@ pub fn iclip(v: c_int, min: c_int, max: c_int) -> c_int {
     }
 }
 
+#[inline]
+pub fn clip_u8(v: u8) -> u8 {
+    clip(v, u8::MIN, u8::MAX)
+}
```

Comment:
This... doesn't seem to do anything? This seems like something that would be used on a wider integer type to clip it into the u8 range (i.e. what `iclip_u8` is doing). When run for a `u8` value it should always be a no-op, right?

Reply:
Hmm, good point.  It was probably clipping a larger int into a `u8`.  I'll get rid of this and just keep the `iclip_u8`, which I think does it correctly.

Reply:
Actually, I think you're right that `iclip_pixel` for `BitDepth8` is just a nop.

Reply:
I've left `clip_u8` now, but as a narrowing conversion from `T` to `u8`.  That turns into a nop for `BitDepth8` where `T = Self::Pixel = u8`, and so I've also merged the `fn iclip_pixel` definitions as they're clearly equivalent now.  I also fixed `fn bitdepth_max` for `BitDepth8`, which was not working for some reason.